src/board_controller/openbci/ganglion_bglib/callbacks.cpp

```diff
@@ -68,8 +71,18 @@ void ble_evt_gap_scan_response (const struct ble_msg_gap_scan_response_evt_t *ms
 
     if (name_found_in_response)
     {
-        // strcasestr is unavailable for windows
-        if (strstr (name, "anglion") != NULL)
+        // Determine the firmware version from the advertised name
+        if (msg->data.data[0] == 20 && msg->data.data[13] == '3')
+        {
+            GanglionLib::firmware = 3;
```

Comment:
I think this entire if block should be under the check for expected name pattern

Reply:
See the note about the serial_number parameter below. If a specific device name is passed to brainflow, the firmware must still be determined. Putting this block outside of the name and serial_number checks prevents duplicating the same logic in both locations.

Reply:
I mean that its probably more reliable to check fw after validating the name, some data from other devices can be send and first they should be filtered by the name. Here you will set fw value wo such filtering

Reply:
or two ganglions in the same place with two different fw versions, wo check for name you can mess up here